### DISCUSSION TITLE
Setup troubles (2)

### DISCUSSION THREAD

**Discussion by ManMead** [meisnate12/Plex-Meta-Manager#1665] [Q&A] [answered]
Hello
I'm trying to setup PMM for the first time, following the Local Walkthrough, and I'm getting stuck at the "Installing Requirements" stage.
When I execute the install command...
`python -m pip install -r requirements.txt `
I get (with some text removed)...

```
Collecting idna<4,>=2.5
  Downloading https://www.piwheels.org/simple/idna/idna-3.4-py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 832 kB/s
Collecting six>=1.7.0
  Downloading https://www.piwheels.org/simple/six/six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting ruamel.yaml.clib>=0.2.7
  Downloading https://www.piwheels.org/simple/ruamel-yaml-clib/ruamel.yaml.clib-0.2.8-cp39-cp39-linux_armv7l.whl (595 kB)
     |████████████████████████████████| 595 kB 1.8 MB/s
Collecting smmap<6,>=3.0.1
  Downloading https://www.piwheels.org/simple/smmap/smmap-5.0.1-py3-none-any.whl (24 kB)
Installing collected packages: urllib3, smmap, idna, charset-normalizer, certifi, six, ruamel.yaml.clib, requests, gitdb, docopt, tmdbapis, schedule, ruamel.yaml, retrying, python-dotenv, psutil, PlexAPI, pillow, pathvalidate, num2words, lxml, GitPython, arrapi
Successfully installed GitPython-3.1.36 PlexAPI-4.15.2 arrapi-1.4.3 certifi-2023.7.22 charset-normalizer-3.3.0 docopt-0.6.2 gitdb-4.0.10 idna-3.4 lxml-4.9.3 num2words-0.5.12 pathvalidate-3.2.0 pillow-10.0.1 psutil-5.9.5 python-dotenv-1.0.0 requests-2.31.0 retrying-1.3.4 ruamel.yaml-0.17.32 ruamel.yaml.clib-0.2.8 schedule-1.2.0 six-1.16.0 smmap-5.0.1 tmdbapis-1.2.0 urllib3-2.0.6
```

...which sort of looks ok to me (?), but in response to...
`python plex_meta_manager.py -r
`
I get... 
`Requirements Error: Requirements are not installed
`

As discussed in the other post with the same title I tried running...
`pip install setuptools
`
...and got the response...
```
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: setuptools in ./pmm-venv/lib/python3.9/site-packages (44.1.1)
```

I am running on a Raspberry Pi 3 with...
```
"Raspbian GNU/Linux 11 (bullseye)
Python 3.9.2
git version 2.30.2

```
I followed the Walthrough up to this point without any errors.
I am a total newbie regarding python so have no idea where to go from here! 
Any ideas?

**Comment by chazlarson** [comment]
That message is displayed if one of these fails:
```
import plexapi, psutil, requests, schedule
from dotenv import load_dotenv
from PIL import ImageFile
from plexapi import server
from plexapi.exceptions import NotFound
from plexapi.video import Show, Season
```

in the PMM directory, with the venv activated, type `python`, hit `return`, then in the python interpreter, copy-paste each of those lines individually to see which one[s] produce an error and cannot be loaded.

I suspect there's a dependency that doesn't support the Pi.

**Comment by chazlarson** [comment]
Also, when you run `python plex_meta_manager.py -r`, the virtual environment is activated, right?

**Reply by ManMead** [reply]
Hi
Thanks for the quick response.

Seems I have a missing component...
```
pi@Apollo:~ $ cd Plex-Meta-Manager
pi@Apollo:~/Plex-Meta-Manager $ source pmm-venv/bin/activate
(pmm-venv) pi@Apollo:~/Plex-Meta-Manager $ python
Python 3.9.2 (default, Mar 12 2021, 04:06:34)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import plexapi, psutil, requests, schedule
>>> from dotenv import load_dotenv
>>> from PIL import ImageFile
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/pi/Plex-Meta-Manager/pmm-venv/lib/python3.9/site-packages/PIL/ImageFile.py", line 35, in <module>
    from . import Image
  File "/home/pi/Plex-Meta-Manager/pmm-venv/lib/python3.9/site-packages/PIL/Image.py", line 82, in <module>
    from . import _imaging as core
ImportError: libopenjp2.so.7: cannot open shared object file: No such file or directory
>>> from plexapi import server
>>> from plexapi.exceptions import NotFound
>>> from plexapi.video import Show, Season
>>>
```
Is this something that can be fixed?

**Reply by chazlarson** [reply]
Not sure.  Perhaps:
<https://www.techcoil.com/blog/how-to-setup-python-imaging-library-pillow-on-raspbian-stretch-lite-for-processing-images-on-your-raspberry-pi/>

Or maybe the first answer here:
<https://stackoverflow.com/questions/20176883/how-to-install-pil-with-jpeg-support-on-a-raspberry-pi>

Or the first answer here based on the error itself:
<https://stackoverflow.com/questions/48012582/pillow-libopenjp2-so-7-cannot-open-shared-object-file-no-such-file-or-directo>

**Reply by ManMead** [reply, marked as answer]
This problem sorted by...
`sudo apt install libopenjp2-7`
Thanks so much for your help Chaz.
Onwards and upwards!
